addons/product/models/product_attribute.py

```diff
@@ -447,7 +447,14 @@ def write(self, values):
                         _("You cannot change the product of the value %s set on product %s.") %
                         (ptav.display_name, ptav.product_tmpl_id.display_name)
                     )
-        return super(ProductTemplateAttributeValue, self).write(values)
+        res = super(ProductTemplateAttributeValue, self).write(values)
+
+        if 'exclude_for' in values:
+            # if last variant and exlusion is not a parent exclusion
+            if self.product_tmpl_id.product_variant_count == 1 and self.product_tmpl_id.id == values['exclude_for'][0][2]['product_tmpl_id']:
+                raise UserError("Forbidden action. You cannot exclude the last variant in this context. Please archive or delete your product directly if intended.")
```

Comment:
```suggestion
            for template in self.product_tmpl_id:
                # if last variant and exlusion is not a parent exclusion
                if template.product_variant_count == 1 and template.id == values['exclude_for'][0][2]['product_tmpl_id']:
                    raise UserError("Forbidden action. You cannot exclude the last variant in this context. Please archive or delete your product directly if intended.")
```
shouldn't this be done by each template? 🤔

Reply:
the `product.template.attribute.value().product_tmpl_id` is a many2one as far as I can see (it is a record only for a given template, not a `product.attribute.value`)

Reply:
but if in self you have two ptavs, each one with a different product_tmpl_id, then self.product_tmpl_id = 2 templates

Reply:
ah yes I missed that, I'm not enough used to `self.product_tmpl_id` working like `self.mapped('product_tmpl_id')`